log/field.go

```diff
@@ -0,0 +1,201 @@
+package log
+
+import "math"
+
+type fieldType int
+
+const (
+	stringType fieldType = iota
+	boolType
+	intType
+	int32Type
+	uint32Type
+	int64Type
+	uint64Type
+	float32Type
+	float64Type
+	errorType
+	objectType
+	lazyLoggerType
```

Comment:
As for arrays, I would say the main reason not to include them is that it cannot be done while preserving type safety without doubling the number of types already present. Array of `interface{}` is fairly meaningless (might as well use `log.Object()`), while `log.Uint32Array()`, `log.StringArray()`, etc. is (to me) unpalatable.